### ISSUE TITLE
Unable to connect to NET-PwrCtrl PRO

### ISSUE THREAD

**Issue by Xyolyp** [dan1-de/ioBroker.anelhut#20]
**Describe the bug**  
I'm unable to get a connection to my NET-PwrCtrl PRO Version PRO 3.0 DE established

**To Reproduce**  
1. Enabled UDP connectivity
2. Oddly I can configure a password specifically for UCP communication but there is seemingly no user linked to it
3. Try all the combinations of users and/or passwords
4. No connection can be established

**Expected behavior**  
I'd love to see some way of establishing a connection to the hardware and controlling all eight outputs.

**Versions:**  
 - Adapter version: 1.0.8
 - JS-Controller version: 4.0.15
 - Node version: 14.19.0
 - Operating system: Debian 10 buster

**Comment by dan1-de** [comment]
You can check if you can recieve the udp broadcast on your debian computer with the following command: 

netcat -vv -l -p **2078** -u

You have to replace 2078 with the port which is shown in the anel webinterface here:
![image](https://user-images.githubusercontent.com/52854781/155876910-837013c2-0350-4f1f-bbbe-9f5c601cecd8.png)

After you have entered the command the computer is listening on udp broadcast traffic. If you change now one of the inputs/outputs of the anel device, you should recieve a message in the command line.

**Comment by Xyolyp** [comment]
Thank you for the quick replies!
This is what my user settings dialog looks like. Notice the "Password for UDP Communication" at the bottom.
![image](https://user-images.githubusercontent.com/13995942/155877958-f1cc55ad-026d-4e0a-bb59-b17b11cc8a7e.png)

Using natcat I was able to receive the following message:
```
$ sudo netcat -vv -l -p 77 -u
listening on [any] 77 ...
192.168.55.48: inverse host lookup failed: Unknown host
connect to [192.168.55.33] from (UNKNOWN) [192.168.55.48] 75
NET-PwrCtrl:BUERO          :192.168.55.48:255.255.255.0:192.168.55.1:0.4.163.10.3.50:Switch Bro 1,1:Monitore auen,1:Nr. 3,0:Laptop,0:Lautsprecher,1:PC,1:Monitor Mitte,1:Tischlampe,0:33:80
```

**Comment by dan1-de** [comment]
Netcat looks good. Can you provide a screenshot of the iobroker anel adapter settings ?

**Comment by Xyolyp** [comment]
Sure, firstly the config:
![image](https://user-images.githubusercontent.com/13995942/155879712-4eca821b-59ef-4b42-82e9-d21598facc2f.png)
And the logs:
![image](https://user-images.githubusercontent.com/13995942/155879751-e328849f-9919-462f-a3c1-5d2c034829d4.png)

**Comment by dan1-de** [comment]
I think I found the issue.
There is some code adaption needed.
I will give you feedback when the adaption was done and send you a url with the new adapter version for testing.

**Comment by dan1-de** [comment]
H @Xyolyp, 

i implemented a bugfix. 
Can you install the adapter from this github url please and give me feedback:
https://github.com/dan1-de/ioBroker.anelhut/tarball/proV3

Version is 1.0.9.

Br, 
Daniel

**Comment by Xyolyp** [comment]
@dan1-de thank you so much reading data works now but writing sadly doesn't.
See the logs below:
![image](https://user-images.githubusercontent.com/13995942/156050906-9b773c18-f42c-4a25-bac6-f0eed8d2c340.png)
If you'd like I can offer you to demonstrate anything over discord, teamviewer or whatever.

**Comment by dan1-de** [comment]
Hi @Xyolyp 
can you try this command in linux shell please:

echo "Sw_on<relaisNumber>adminanel" | nc -4u <yourIP> 75

for example: 
echo "Sw_on2adminanel" | nc -4u 10.10.10.10 75

Br, 
Daniel

**Comment by Xyolyp** [comment]
Hi @dan1-de ,

Sadly I had to remove the `4` from all commands as nc was telling me it is no valid parameter. None of the combinations I executed resulted in any change to any relais.

I'll shortly send an email your way. 
Best regards,
Jonathan

**Comment by Xyolyp** [comment]
After performing a firmware upgrade to the latest version published for my hardware (Firmware 4.2 for Hardware 3.0) the adapter seems to work as intended. I got the firmware update from: https://de.anel.eu/index.htm?src=/usb-deutsch/Update_FW_NET/firmware_net-pwrctrl.html and had to disable my desktops firewall to perform the firmware upgrade.
